### ISSUE TITLE
Bug: Miner crash caused by concurrent map read/write in 0chain.net/chaincore/node/node.go

### ISSUE THREAD

**Issue by bbist** [0chain/0chain#145]
Miner crashed with following panic.
```
miner_1       | fatal error: concurrent map read and map write
miner_1       | 
miner_1       | goroutine 1255468 [running]:
miner_1       | runtime.throw(0xfd92d1, 0x21)
miner_1       | 	/usr/local/go/src/runtime/panic.go:617 +0x72 fp=0xc000061560 sp=0xc000061530 pc=0x449672
miner_1       | runtime.mapaccess2_faststr(0xede3a0, 0xc002f6b110, 0xfd9fff, 0x22, 0xc000061790, 0x526c46)
miner_1       | 	/usr/local/go/src/runtime/map_faststr.go:116 +0x7a2 fp=0xc000061600 sp=0xc000061560 pc=0x42c592
miner_1       | 0chain.net/chaincore/node.(*Node).getTimer(0xc002f2ab40, 0xfd9fff, 0x22, 0x0, 0x0)
miner_1       | 	/0chain.net/miner/miner/vendor/0chain.net/chaincore/node/node.go:383 +0x8a fp=0xc000061700 sp=0xc000061600 pc=0xb88a4a
miner_1       | 0chain.net/chaincore/node.(*Node).GetTimer(0xc002f2ab40, 0xfd9fff, 0x22, 0x0, 0x0)
miner_1       | 	/0chain.net/miner/miner/vendor/0chain.net/chaincore/node/node.go:379 +0x9b fp=0xc000061758 sp=0xc000061700 pc=0xb8896b
...
```
Issue was that the maps **TimersByURI** and **SizeByURI** were not properly locked while accessing. Problems were in following methods:
1. **updateSendMessageTimings()**: called in **updateMessageTimings()** called in worker.go's **statusUpdate()** and **statusMonitor()** methods without explicit locks.
2. **updateRequestMessageTimings()**: same as above.
3. **SetNodeInfo()**

**FIX**
Locking appropriate mutex. 
In this case **mutex** field present in struct **Node** while accessing either **TimersByURI** or **SizeByURI** maps.

**Comment by mallochine** [comment]
Hm, usually concurrency problems in Golang are solved using channels to a goroutine who will be the sole reader/writer of those two maps...these are usually a lot easier to work/reason with than mutexes.

**Comment by bbist** [comment]
@mallochine 
I agree, in general. It was just a quick fix for existing code base. Here are the changes: https://github.com/0chain/0chain/commit/246c3f2368a38b3f4e8dd78c4d59d08a7dd54d62.

**Comment by peterlimg** [comment]
Fixed by #148
